test/should_pass/list_exhaustiveness_checking_regressions.erl

```diff
@@ -0,0 +1,17 @@
+-module(list_exhaustiveness_checking_regressions).
+
+-export([f/2, g/1, h/1]).
+
+f( Needle, [Needle | _]) -> ok;
+f( Needle, [_ | Haystack]) -> f(Needle, Haystack);
+f(_Needle, []) -> ok.
+
+-spec g([a | b]) -> ok.
+g([a | _Haystack]) -> ok;
+g([_ | Haystack]) -> g(Haystack);
+g([]) -> ok.
```

Comment:
Good test cases.

Without the 2nd or 3rd clause, do we detect the unexhaustiveness? If yes, are there should-fail tests for this already?

Reply:
Hmmm, good point, I don't think there are explicit tests for this.

BTW, we don't detect anything in the case of `f`, but it's because it lacks a spec. However, it's like that intentionally, since we were getting the regressions (false positives) on such spec-less functions when running `make gradualize`.

Reply:
There are now in `test/should_fail/exhaustive_list_variants.erl`.